virtool/users/sessions.py

```diff
@@ -131,42 +137,42 @@ async def create_reset_code(
     """
     reset_code = secrets.token_hex(32)
 
-    await db.sessions.update_one(
-        {"_id": session_id},
-        {
-            "$set": {
-                "reset_code": reset_code,
-                "reset_remember": remember,
-                "reset_user_id": user_id,
-            }
-        },
-    )
+    session = json.loads(await redis.get(session_id))
+    session_expiry = await redis.ttl(session_id)
```

Comment:
Use `asyncio.gather` here to make requests concurrently.

Extra context from `session_` is not required IMO.

Reply:
> Use `asyncio.gather` here to make requests concurrently.
Done, good call.

> Extra context from `session_` is not required IMO.
Not sure what you mean here? The limitations of `redis.set()` mean that the whole json string is set every time, so I need the full session data. I could probably prevent this by switching to storing the data in a redis hash, but that will require writing a custom decoder encoder for the non-compatible datatypes. Certainly doable and I can swtich course, but last we talked the plan was to stick with json.